src/logplex_message.erl

```diff
@@ -139,7 +212,37 @@ process_redis(ChannelId, ShardInfo, Msg, _Flag) ->
                                       Msg, Expiry),
     logplex_queue:in(BufferPid, Cmd).
 
-parse_msg(Msg) ->
+%% @doc Takes a list of messages and adds them in a batch to redis log buffers
+%% under the given channel id.
+process_redis_batch(_ChannelId, _Msgs, Flag) when Flag =:= no_redis;
+                                                  Flag =:= true ->
+    ok;
+process_redis_batch(ChannelId, Msgs, _Flag) when is_list(Msgs) ->
+    ShardInfo = shard_info(),
+    Expiry = logplex_app:config(redis_buffer_expiry),
+    HistorySize = list_to_binary(integer_to_list(logplex_app:config(log_history))),
+    {Map, Interval} = logplex_shard_info:map_interval(ShardInfo),
+    BufferPid = logplex_shard:lookup(binary_to_list(ChannelId),
+                                     Map, Interval),
+    Cmd = redis_helper:build_push_batch_msgs(ChannelId, HistorySize,
+                                             Msgs, Expiry),
+    BatchSize = length(Msgs),
+    logplex_queue:in(BufferPid, {Cmd, BatchSize}).
```

Comment:
@voidlock Certainly possible to change the code according to your proposal from above. I've kept the new code close to the old path which is why this is what it is 😉.